zerver/views/users.py

```diff
@@ -572,7 +571,10 @@ def get_user_data(
     compress very poorly compared to other data.
     """
     realm = user_profile.realm
-    if realm.email_address_visibility != Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE:
+    if (
+        target_user is not None
+        and target_user.email_address_visibility != UserProfile.EMAIL_ADDRESS_VISIBILITY_EVERYONE
+    ):
```

Comment:
I think `get_raw_user_data` has a copy of this logic already; perhaps we can do a prep commit deleting this block?  https://github.com/zulip/zulip/pull/20102/commits/254b578f35b65a1acc811174252e1708aa05e691#r865306423

Reply:
Removed this in the same commit only since the code in `get_raw_user_data` is added that commit itself.